### PR TITLE
HPCC-16735 Refactoring to create MemberFunction objects

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

### CODE REVIEW

**Comment by ghalliday** [comment]
Richard, this is targeting master, but could be retargeted to 6.4 if we consider issue #16734 significant enough to try and fix in 6.4.
@shamser please review

File: ecl/hqlcpp/hqlhtcpp.cpp [hunk 1]
```diff
@@ -17062,98 +17064,99 @@ ABoundActivity * HqlCppTranslator::doBuildActivityTempTable(BuildCtx & ctx, IHql
...
-            for (row = 0; row < maxRows; row++)
+            buildAssign(func.ctx, self, values);
+            buildReturnRecordSize(func.ctx, selfCursor);
+            rowsExpr.setown(getSizetConstant(1));
+        }
+        else if ((values->getOperator() == no_list) || (values->getOperator() == no_null))
+    //  else if (((values->getOperator() == no_list) &&
+    //                  (!values->isConstant() || (values->queryType()->queryChildType()->getSize() == UNKNOWN_LENGTH))) || (values->getOperator() == no_null))
+        {
+            //
```
Comment (shamser) [inline ecl/hqlcpp/hqlhtcpp.cpp]: @ghalliday Any reason that these comment lines haven't been removed?

**Review by shamser — COMMENTED** [review]
@ghalliday Couple of minor comments/

File: ecl/hqlcpp/hqlhtcpp.cpp (hunk 1 above)
Comment (ghalliday) [inline ecl/hqlcpp/hqlhtcpp.cpp]: No reason, but I wouldn't normally clean up deleting lines as part of an unrelated PR.

**Comment by ghalliday** [comment]
@richardkchapman added/removed comments.
This should continue to target master rather than 6.4 because the previous idea isn't a viable approach for fixing #16734

**Pull request merged**